src/http/http2_types.h

```diff
@@ -18,9 +18,11 @@
 namespace http2
 {
   using StreamId = int32_t;
+  constexpr static StreamId DEFAULT_STREAM_ID = 0;
+
   using StreamCloseCB = http::StreamOnCloseCallback;
 
-  constexpr static size_t max_data_read_size = 1 << 20;
+  constexpr static size_t max_frame_size = 1 << 14;
```

Comment:
Taking this down to the default of 16kB as [one the test cases](https://github.com/summerwind/h2spec/blob/70ac2294010887f48b18e2d64f5cccd48421fad1/client/4_2_frame_size.go#L106) in `h2spec` does not seem to take the connection maximum frame size value into account and instead solely depends on the maximum allowed header length on this connection. This can be specified to the `h2spec` CLI and I'll address this next when I add support for maximum header sizes, etc. for HTTP/2.